Source/WebInspectorUI/UserInterface/Views/FontDetailsPanel.js

```diff
@@ -73,21 +99,77 @@ WI.FontDetailsPanel = class FontDetailsPanel extends WI.StyleDetailsPanel
         this._fontAdditionalFeaturesGroup.rows = featureRows;
         this._fontAdditionalFeaturesGroup.hidden = !featureRows.length;
 
-        let variationRows = [];
-        for (let [key, value] of this._fontVariationsMap) {
-            let name = value.name ? WI.UIString("%s (%s)").format(value.name, key) : key;
-            variationRows.push(new WI.DetailsSectionSimpleRow(name, this._formatVariationValue(value)));
+        // Variation properties
+        for (let [tag, fontVariationAxis] of this._fontVariationsMap) {
+            let variationRow = this._fontVariationRowsMap.get(tag);
+            variationRow.value = fontVariationAxis.value ?? fontVariationAxis.defaultValue;
+            variationRow.warningMessage = (fontVariationAxis.value < fontVariationAxis.minimumValue || fontVariationAxis.value > fontVariationAxis.maximumValue) ? WI.UIString("Axis value outside of supported range: %s – %s", "A warning that is shown in the Font Details Sidebar when the value for a variation axis is outside of the supported range of values").format(this._formatAxisValueAsString(fontVariationAxis.minimumValue), this._formatAxisValueAsString(fontVariationAxis.maximumValue)) : null;
         }
-        if (!variationRows.length) {
+
+        if (!this._fontVariationRowsMap.size) {
             let emptyRow = new WI.DetailsSectionRow(WI.UIString("No additional variation axes.", "No additional variation axes. @ Font Details Sidebar", "Message shown when there are no additional variation axes to show."));
             emptyRow.showEmptyMessage();
-            variationRows.push(emptyRow);
+
+            this._fontVariationsGroup.rows = [emptyRow];
         }
-        this._fontVariationsGroup.rows = variationRows;
     }
 
     // Protected
 
+    detached()
+    {
+        super.detached();
+
+        this._fontStyles = null;
+        this._abortController?.abort();
+        this._abortController = null;
+    }
+
+    layout()
+    {
+        if (this.layoutReason === WI.View.LayoutReason.Resize)
+            return;
+
+        if (!this.nodeStyles.computedStyle || !this.nodeStyles.computedPrimaryFont) {
+            this._fontStyles = null;
+            return;
+        }
+
+        this._abortController?.abort();
+        this._abortController = new AbortController();
+        this._fontStyles = new WI.FontStyles(this.nodeStyles);
+
+        this._fontPropertiesMap = this._fontStyles.propertiesMap;
+        this._fontVariationsMap = this._fontStyles.variationsMap;
+        this._fontFeaturesMap = this._fontStyles.featuresMap;
```

Comment:
Do we actually need to have local members here? Maybe we use some private getters instead if we don't want to type `_fontStyles` every time we edit?

Reply:
Took your suggestion and went ahead making them actual private getters:

```
get #fontPropertiesMap()
{
    return this._fontStyles?.propertiesMap ?? new Map;   
}

get #fontVariationsMap()
{
    return this._fontStyles?.variationsMap ?? new Map;
}

get #fontFeaturesMap()
{
    return this._fontStyles?.featuresMap ?? new Map;
}
```